core/src/main/java/org/elasticsearch/transport/RequestHandlerRegistry.java

```diff
@@ -52,8 +55,21 @@ public Request newRequest() {
             return requestFactory.get();
     }
 
-    public TransportRequestHandler<Request> getHandler() {
-        return handler;
+    public void processMessageReceived(Request request, TransportChannel channel) throws Exception {
+        final Task task = taskManager.register(channel.getChannelType(), action, request);
+        if (task == null) {
+            handler.messageReceived(request, channel);
+        } else {
+            boolean success = false;
```

Comment:
I know this is a pattern we use everywhere but I don't like the name `success` to mean "I've forked this request and now it's the listener's problem" but I don't know a better name.